core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -1792,7 +1792,9 @@ public void delete(AlluxioURI path, DeleteContext context)
             throw e;
           }
         }
-        mMountTable.checkUnderWritableMountPoint(path);
+        if (!context.getOptions().getAlluxioOnly()) {
+          mMountTable.checkUnderWritableMountPoint(path);
```

Comment:
@yuzhu Do you think we should allow rm alluxioOnly files when a mount point is read-only?

Reply:
@jiacheliu3 We have a scenario where the underfs is read-only and the underfs's data changes. When I execute ./alluxio fs checkConsistency -r /, there are some inconsistent files that should be deleted cannot be deleted.

Reply:
Another case, ./alluxio fs rm --alluxioOnly /test , the file cannot  be removed too.

Reply:
But what if the UFS file does not change and you just delete the file in Alluxio? The change will not propagate the UFS and will not be rejected. Later the file will appear again due to sync. This will be confusing.

Reply:
@jiacheliu3 Yeah, the purpose of deleting the file is that the file was never be consistent with the ufs, so we want to delete it from Alluxio, and then, the new consistent inode will be created.

Reply:
Thanks for the explanation, makes sense to me

Reply:
I'm actually a little concerned about files that are not in UFS. For example you `--alluxioOnly`  remove an ASYNC_THROUGH file before it's persisted, or remove a MUST_CACHE file. These can all lead to data loss.

What about we only allow `--alluxioOnly` remove a file only if it's persisted? @yuzhu @maobaolong